app/src/main/java/org/simple/clinic/forgotpin/confirmpin/ForgotPinConfirmPinScreenController.kt

```diff
@@ -67,8 +71,57 @@ class ForgotPinConfirmPinScreenController @Inject constructor(
         .map { { ui: Ui -> ui.showPinMismatchedError() } }
   }
 
-  private fun hideErrors(events: Observable<UiEvent>): Observable<UiChange> {
+  private fun hideErrorsOnPinTextChanged(events: Observable<UiEvent>): Observable<UiChange> {
     return events.ofType<ForgotPinConfirmPinTextChanged>()
         .map { { ui: Ui -> ui.hideError() } }
   }
+
+  private fun handlePinSubmitted(events: Observable<UiEvent>): Observable<UiChange> {
```

Comment:
This function seems to be doing a lot of things. Can this be split into multiple ui change functions?

For start, I at least think that the function name does not indicate its functionality. They are supposed to indicate what they do and not what events they consume.

Reply:
A quick scan of this function suggests that showing of validation errors can be moved to a different function.

Reply:
Will fix in the next PR.